app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/resources/ResourcesScreen.kt

```diff
@@ -83,13 +101,54 @@ private fun ShowResourcesScreenContent(
                 stringResource(id = R.string.empty_state_description_resource)
             )
         } else {
-            Spacer(modifier = Modifier.height(dimensionResource(id = R.dimen.dimension_8dp)))
+            SearchBar(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(padding)
+                    .padding(dimensionResource(id = R.dimen.dimension_16dp)),
+                query = searchText,
+                onQueryChange = { searchText = it },
+                onSearch = {
+                    active = false
+                    viewModel.getSearchingResult(it)
+                },
+                active = active,
+                onActiveChange = { active = it },
+                placeholder = { Text(text = stringResource(id = R.string.search_bar_default_text)) },
+                leadingIcon = {
+                    Icon(
+                        imageVector = Icons.Default.Search,
+                        contentDescription = null
+                    )
+                },
+                trailingIcon = {
+                    if (active) {
+                        Icon(
+                            modifier =
+                            Modifier.clickable {
+                                if (searchText.isNotEmpty()) {
+                                    searchText = ""
+                                } else {
+                                    active = false
+                                }
+                            },
+                            imageVector = Icons.Default.Close,
+                            contentDescription = null
+                        )
+                    }
+                },
+            ) {
+            }
             LazyColumn(
-                modifier = Modifier.padding(padding),
+                modifier = Modifier
+                    .padding(padding)
+                    .padding(top = 76.dp),
```

Comment:
I am not sure if it is the best practice, but since the expanding sheet of the search bar is an infinitely vertical scrolling sheet, if I include SearchBar in the LazyColum, the error will raise for two infinitely vertically scrolling components. Then I added this fixed padding to give the search bar enough space outside of the lazy column.

Reply:
What if you add the search bar outside lazy column and wrap within a Column ?

Reply:
Let me try

Reply:
Yes, your suggested implementation works out.

https://github.com/Audienix/InterPrep/assets/67870437/1ed6d745-9ff2-42e1-a858-47e8a9dbb9e2

Reply:
Nice, glad it worked. :)

Reply:
I have pushed this change.